ciao-controller/client.go

```diff
@@ -114,26 +109,48 @@ func (client *ssntpClient) releaseResources(instanceID string) error {
 	return nil
 }
 
-func (client *ssntpClient) removeInstance(instanceID string) {
-	err := client.releaseResources(instanceID)
+func (client *ssntpClient) removeInstance(i *types.Instance) {
+	tenant, err := client.ctl.ds.GetTenant(i.TenantID)
+	if err != nil {
+		glog.Warningf("Error retrieving tenant %v", err)
+		return
+	}
+
+	err = client.releaseResources(i)
 	if err != nil {
 		glog.Warningf("Error when releasing resources for deleted instance: %v", err)
```

Comment:
@rbradford going back and reviewing the original code here, it looks like it was intentionally not returning in order to attempt to try to complete as much of the removal as could be completed. I think that makes sense to keep.